### ISSUE TITLE
Enhance Model Fusion Strategy for Multi-Modal Features

### ISSUE THREAD

**Issue by Fantasyzyh** [hongzhonglu/DeepEnzyme#8]
## Current Issue
The current model uses three feature modalities (molecule features, fp1, fp2) that perform well individually but degrade when combined. The current fusion strategy using channel attention may not be capturing the complex interactions between these modalities effectively.

## Proposed Solution
Design a new fusion architecture with the following improvements:

### 1. Cross-Modal Interactive Fusion
```python
class CrossModalFusion(nn.Module):
    def __init__(self, hidden_dim):
        super().__init__()
        self.cross_attn = nn.MultiheadAttention(hidden_dim, num_heads=8)
        self.norm = nn.LayerNorm(hidden_dim)
        self.ffn = nn.Sequential(
            nn.Linear(hidden_dim, 4 * hidden_dim),
            nn.ReLU(),
            nn.Linear(4 * hidden_dim, hidden_dim)
        )
        
    def forward(self, x1, x2):
        # Cross attention between modalities
        attended, _ = self.cross_attn(x1, x2, x2)
        x1 = self.norm(x1 + attended)
        return self.norm(x1 + self.ffn(x1))
```

### 2. Hierarchical Feature Integration
Instead of fusing all modalities at once, use a hierarchical approach:
1. First fuse molecule features with fp1
2. Then fuse the result with fp2
3. Use residual connections to preserve important individual features

```python
class HierarchicalFusion(nn.Module):
    def __init__(self, hidden_dim):
        super().__init__()
        self.fusion1 = CrossModalFusion(hidden_dim)
        self.fusion2 = CrossModalFusion(hidden_dim)
        self.gate = nn.Sequential(
            nn.Linear(3 * hidden_dim, 3),
            nn.Softmax(dim=-1)
        )
        
    def forward(self, mol, fp1, fp2):
        # First level fusion
        fused1 = self.fusion1(mol, fp1)
        
        # Second level fusion
        fused2 = self.fusion2(fused1, fp2)
        
        # Adaptive gating for residual connections
        gate_weights = self.gate(torch.cat([mol, fp1, fp2], dim=-1))
        residual = gate_weights[:, 0:1] * mol + \
                  gate_weights[:, 1:2] * fp1 + \
                  gate_weights[:, 2:3] * fp2
                  
        return fused2 + residual
```

### 3. Uncertainty-Aware Feature Weighting
Add uncertainty estimation for each modality to better weight their contributions:

```python
class UncertaintyWeighting(nn.Module):
    def __init__(self, hidden_dim):
        super().__init__()
        self.uncertainty = nn.Parameter(torch.ones(3))
        
    def forward(self, features):
        weights = F.softmax(-self.uncertainty, dim=0)
        return sum(w * f for w, f in zip(weights, features))
```

### Implementation Steps
1. Replace the current `ChannelAttention` module with the new fusion architecture
2. Modify the forward pass in `KcatRegressor` to use hierarchical fusion
3. Add uncertainty weighting for final feature combination
4. Maintain residual connections throughout to preserve individual modality information

### Additional Recommendations
1. Add layer normalization after each fusion step
2. Consider using FiLM conditioning layers for better feature interaction
3. Implement gradient checkpointing to handle increased memory usage

## Expected Benefits
1. Better preservation of individual modality strengths
2. More sophisticated cross-modal interactions
3. Adaptive weighting based on feature quality
4. Improved gradient flow through hierarchical structure

## Testing Plan
1. Compare performance with original model
2. Analyze contribution of each modality using ablation studies
3. Measure computational overhead
4. Validate on different subsets of the data to ensure robustness
